src/node_crypto.cc

```diff
@@ -1025,6 +1066,46 @@ void SecureContext::LoadPKCS12(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+#ifndef OPENSSL_NO_ENGINE
+void SecureContext::SetClientCertEngine(
+  const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args.Length() == 1 && args[0]->IsString());
+
+  SecureContext* sc = Unwrap<SecureContext>(args.This());
+
+  ClearErrorOnReturn clear_error_on_return;
+
+  // If an engine was previously set, free it.
+  // This is because SSL_CTX_set_client_cert_engine does not itself
+  // 'support' multiple calls by cleaning up before overwriting the
+  // client_cert_engine internal context variable
+  if (sc->ctx_->client_cert_engine != NULL) {
```

Comment:
I'm inclined to say it should throw a JS exception instead of trying to silently fix up.  I mean, when would you change the engine during the lifetime of a `SSL_CTX` anyway?